paywall/src/providers/Web3ProxyProvider.js

```diff
@@ -42,13 +42,13 @@ export default class Web3ProxyProvider {
     // this is the postMessage version of the callback for sendAsync
     addHandler(POST_MESSAGE_WEB3, web3Result => {
       if (
-        !web3Result.hasOwnProperty('error') ||
+        !web3Result.hasOwnProperty('error') &&
         !web3Result.hasOwnProperty('result')
       ) {
         return
       }
-      const { result, error } = web3Result
-      const id = result.id
+      const { result, error = null } = web3Result
```

Comment:
Can L50 and L51 be merged?

Reply:
duh - they needed to be separate when I was handling things the odd way above, good catch